### PR TITLE
Respect excluded fields setting

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/DbAl/WordpressContentEntities/EntityAbstract.php [hunk 1]
```diff
@@ -303,20 +303,7 @@ protected function entityNotFound($type, $guid)
...
-        $nonCloneFields = $this->getNonClonableFields();
-
-        $myFields = $this->toArray();
-        if (is_array($nonCloneFields) && 0 < count($nonCloneFields)) {
-            foreach ($nonCloneFields as $field) {
-                unset ($myFields[$field]);
-            }
-        }
-        $this->resultToEntity($myFields);
-    }
```
Comment (vsolovei-smartling) [inline inc/Smartling/DbAl/WordpressContentEntities/EntityAbstract.php:319]: I strongly believe this function is useless, but interferes with normal testing flow

File: inc/Smartling/Settings/ConfigurationProfileEntity.php [hunk 2]
```diff
@@ -286,11 +286,22 @@ public function setFilterFieldNameRegexp($value)
...
     {
         return $this->stateFields['filter_skip'];
     }
 
+    /**
+     * @return array
+     */
+    public function getFilterSkipArray()
+    {
+        return array_map('trim', explode(PHP_EOL, $this->stateFields['filter_skip']));
```
Comment (PavelLoparev) [inline inc/Smartling/Settings/ConfigurationProfileEntity.php:302]: `return array_map('trim', explode(PHP_EOL, $this->getFilterSkip()));`

File: inc/config/services.yml [hunk 3]
```diff
@@ -235,6 +235,8 @@ services:
       - [ "setSiteHelper", [ "@site.helper" ]]
 
   content.helper:
+    arguments:
+      - "@wp.proxy"
```
Comment (PavelLoparev) [inline inc/config/services.yml:239]: Do we have places where we manually instantiate this object (except tests)? We need to pass proxy there as well if any

File: inc/Smartling/Submissions/SubmissionEntity.php [hunk 4]
```diff
@@ -841,11 +841,32 @@ public function setBatchUid($batchUid)
...
+     */
     public function getLockedFields()
     {
         return $this->stateFields[static::FIELD_LOCKED_FIELDS];
     }
 
+    /**
+     * @return array
+     */
+    public function getLockedFieldsArray()
```
Comment (PavelLoparev) [inline inc/Smartling/Submissions/SubmissionEntity.php:857]: I don't see where you use this function

**Review by PavelLoparev — APPROVED** [review]
Left some comments/questions that I want to clarify before merging

File: inc/Smartling/DbAl/WordpressContentEntities/EntityAbstract.php (hunk 1 above)
Comment (PavelLoparev) [reply inc/Smartling/DbAl/WordpressContentEntities/EntityAbstract.php:319]: The function looks ok to me - just returns new object without some fields. Questions:
1. Where in code we use `clone`? I'm afraid after this change we will brake some flow(s)
2. Could you provide example on how it interferes testing flow?

**Review by PavelLoparev — CHANGES_REQUESTED** [review]
Approved request by incident last time

File: inc/Smartling/DbAl/WordpressContentEntities/EntityAbstract.php (hunk 1 above)
Comment (vsolovei-smartling) [reply inc/Smartling/DbAl/WordpressContentEntities/EntityAbstract.php:319]: __clone should not return something, but rather change cloned object properties. This is not happening, as resultToEntity creates a new object, alters it's properties and returns it (to be discarded in __clone).
Cloning happens multiple times in code, mostly while reading content, a clone of entity is returned.
When cloning a PhpUnitMockObject, it's mocked functions start returning nulls.

File: inc/config/services.yml (hunk 3 above)
Comment (vsolovei-smartling) [reply inc/config/services.yml:239]: No, everything is handled, and should be handled by DI

File: inc/Smartling/Submissions/SubmissionEntity.php (hunk 4 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Submissions/SubmissionEntity.php:857]: Removed

File: inc/Smartling/Settings/ConfigurationProfileEntity.php (hunk 2 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Settings/ConfigurationProfileEntity.php:302]: Chanded

File: inc/Smartling/DbAl/WordpressContentEntities/EntityAbstract.php (hunk 1 above)
Comment (vsolovei-smartling) [reply inc/Smartling/DbAl/WordpressContentEntities/EntityAbstract.php]: Reverted to reintroduce change in a future pr

**Review by PavelLoparev — APPROVED** [review]

File: readme.txt [hunk 5]
```diff
@@ -58,6 +58,9 @@ Additional information on the Smartling Connector for WordPress can be found [he
 3. Track translation status within WordPress from the Submissions Board. View overall progress of submitted translation requests as well as resend updated content.
 
 == Changelog ==
+= 1.15.1 =
+* Auto synchronize properties on translated page with source feature doesn't alter locked and excluded fields
```
Comment (dimitrystd) [inline readme.txt:62]: It's a little bit hard to read when you don't know\remember feature names.

---

The feature `Auto synchronize properties on translated page with source` now doesn't alter locked and excluded fields on the target page

**Review by dimitrystd — COMMENTED** [review]
@vsolovei-smartling I added desired update for readme. Will push it with the next release.
Also, it will be nice to add new unit tests instead of adding complexity to existing. It can be not clear how assert is important during refactoring in the far future.

**Pull request merged**